### DISCUSSION TITLE
This image is reported as affected by CVE-2024-24790

### DISCUSSION THREAD

**Discussion by pranjalthakur98** [docker-library/rabbitmq#733] [General]
We are using Rabbitmq as a base image to develop application on the top of it but we are tagged with a critical vulnerability of Go (golang) [CVE-2024-24790⁠]. This is stopping us from using it. 
Can this vulnerability could be fixed ? 
We could see this is only with 4.* version of rabbitmq image with ubuntu OS.

**Comment by michaelklishin** [comment]
@pranjalthakur98 this image does not ship any Go binaries. Therefore it cannot be affected by CVE-2024-24790: no Go-based servers are started as part of this image.

**Comment by michaelklishin** [comment]
@pranjalthakur98 RabbitMQ is not a Go-based tool, so I do not see how your claim can be true. The title is therefore making a bold claim without providing any evidence of a vulnerability in RabbitMQ.

[CVE-2024-24790](https://nvd.nist.gov/vuln/detail/CVE-2024-24790) details in the NIST database further confirm that the issue is in the Go IP/socket implementation, so any Go tooling involved at build time would not affect this image or RabbitMQ.

Please post some evidence or this won't be taken seriously.

**Comment by michaelklishin** [comment]
@tianon @yosifkit while we don't have any details or proof here to work with, as a general question,
what is the official image policy w.r.t. Go vulnerabilities? Go is so heavily used throughput the OCI tooling space that I can see how a `net/netip` or `archive/zip` vulnerability can be "attributed" to an image by automatic scanners, even though no started processes are actually Go-based.

This is not the first "issue report" to stem from an automated scan and it won't be the last, so I'd like to understand how official images in general approach this problem.

**Reply by tianon** [reply]
I'm guessing this is related to the `gosu` binary, and it's most certainly _not_ "vulnerable" to CVE-2024-24790: https://github.com/tianon/gosu/blob/HEAD/SECURITY.md (the code for `gosu` doesn't invoke any of `net/http`, `archive/zip`, etc)

https://github.com/docker-library/rabbitmq/blob/4d384714be0c4e4c528745136c38b7f89620410f/4.0/ubuntu/Dockerfile#L254

FWIW, I have run `govulncheck` on the `gosu` binary from the `rabbitmq` image, and it gets really confused and lists a bunch of things that _can't_ possibly be accurate, and I'm not actually sure why it does.  The only thing different about it is that it comes from Ubuntu's package and is built against a newer version of Go than the ones I publish, which should be _better_ in this respect, not worse.

We also usually link to https://github.com/docker-library/faq#why-does-my-security-scanner-show-that-an-image-has-cves to help explain why security scanners might say one thing or another, and why it's almost never the full story.

**Reply by tianon** [reply]
Oh, I bet the Ubuntu builds strip the binary, so `govulncheck` can't know the symbols for sure: https://github.com/tianon/gosu/blob/4233b796eeb3ba76c8597a46d89eab1f116188e2/Dockerfile#L11

**Reply by michaelklishin** [reply]
CVE scanners that produce false positives is something very common to see. Thank you for taking a closer look. https://github.com/docker-library/faq#why-does-my-security-scanner-show-that-an-image-has-cves looks very relevant indeed.

**Reply by tianon** [reply]
It'll take a long time before it trickles down to an Ubuntu release, but I'm planning to apply https://salsa.debian.org/go-team/packages/gosu/-/commit/84e4d54dff129d01a71efd6d2823a7ffdfcef9dc in Debian to explicitly combat this problem (of `govulncheck` not being able to scan the binary properly/accurately).

**Reply by michaelklishin** [reply]
Sounds great, thank you so much!

**Comment by michaelklishin** [comment]
## Update

Community OCI maintainers explain that `gosu` is included into the image and [`gosu` is not rebuilt to address vulnerabilities that do not affect it](https://github.com/tianon/gosu/blob/master/SECURITY.md).

### `gosu` Does not Use Any Networking Libraries and Provides an "Actual Use-based" Scanner

`gosu` includes a [script that provides a more accurate picture](https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck) based on the analysis of the functions that are **actually used** in the tool. The tool does not use HTTP or communicate over the network in general, for example, it is entirely local tool that, ironically, exists for improving security of the RabbitMQ node process in the image.

So nothing in the RabbitMQ community OCI uses those affected libraries. The image is open source, so anyone can verify this claim with a reasonable amount of effort.
